tests/phpunit/CRM/Contribute/BAO/ContributionTest.php

```diff
@@ -1179,88 +1179,90 @@ public function testgetSalesTaxFinancialAccounts() {
   }
 
   /**
-   * Test for function createProportionalEntry().
+   * Test for function createProportionalFinancialEntries().
    *
    * @param string $thousandSeparator
    *   punctuation used to refer to thousands.
    *
    * @dataProvider getThousandSeparators
    */
-  public function testCreateProportionalEntry($thousandSeparator) {
+  public function testCreateProportionalFinancialEntries($thousandSeparator) {
     $this->setCurrencySeparators($thousandSeparator);
     list($contribution, $financialAccount) = $this->createContributionWithTax();
+
     $params = array(
+      'contribution_id' => $contribution['id'],
       'total_amount' => 55,
       'to_financial_account_id' => $financialAccount->financial_account_id,
       'payment_instrument_id' => 1,
       'trxn_date' => date('Ymd'),
       'status_id' => 1,
-      'entity_id' => $contribution['id'],
     );
-    $financialTrxn = $this->callAPISuccess('FinancialTrxn', 'create', $params);
-    $entityParams = array(
-      'contribution_total_amount' => $contribution['total_amount'],
-      'trxn_total_amount' => 55,
-      'line_item_amount' => 100,
-    );
-    $previousLineItem = CRM_Financial_BAO_FinancialItem::getPreviousFinancialItem($contribution['id']);
+    $financialTrxnID = $this->callAPISuccess('FinancialTrxn', 'create', $params)['id'];
+
+    $contributionStatus = $this->callAPISuccessGetSingle('Contribution', ['id' => $contribution['id']])['contribution_status'];
+    CRM_Contribute_BAO_Contribution::assignProportionalLineItems(['contribution_id' => $contribution['id']], $financialTrxnID, $contribution['total_amount'], $contributionStatus);
+
+    list($financialItems, $taxItems) = CRM_Contribute_BAO_Contribution::getLastFinancialItems($contribution['id']);
     $eftParams = array(
       'entity_table' => 'civicrm_financial_item',
-      'entity_id' => $previousLineItem['id'],
-      'financial_trxn_id' => (string) $financialTrxn['id'],
+      'entity_id' => $financialItems[1]['financial_item_id'],
+      'financial_trxn_id' => $financialTrxnID,
     );
-    CRM_Contribute_BAO_Contribution::createProportionalEntry($entityParams, $eftParams);
     $trxnTestArray = array_merge($eftParams, array(
       'amount' => '50.00',
     ));
     $this->callAPISuccessGetSingle('EntityFinancialTrxn', $eftParams, $trxnTestArray);
   }
 
   /**
-   * Test for function createProportionalEntry with zero amount().
+   * Test for function createProportionalFinancialEntries with zero amount().
    *
    * @param string $thousandSeparator
    *   punctuation used to refer to thousands.
    *
    * @dataProvider getThousandSeparators
    */
-  public function testCreateProportionalEntryZeroAmount($thousandSeparator) {
+  public function testCreateProportionalFinancialEntriesZeroAmount($thousandSeparator) {
```

Comment:
wasn't this test written if the there is payment of $0 then items are allocated proportionally rather updating the contribution to 0?